File: base/src/main/java/me/shetj/base/tools/image/SingletonManager.java
package me.shetj.base.tools.image;

import java.util.HashMap;
import java.util.Map;

public class SingletonManager{
    private static Map<String,Object> objMap = new HashMap<>();
    
    private SingletonManager(){}
    
    public static void registerService(String key,Object instance){
        if(!objMap.containsKey(key)){
            objMap.put(key,instance);
        }
    }
    
    public static Object getService(String key){
        return objMap.get(key);
    }
}

File: baseKit/src/main/java/me/shetj/base/tools/app/SingletonManager.kt
package me.shetj.base.tools.app

import androidx.annotation.Keep

@Keep
object SingletonManager {
    private val objMap = HashMap<String, Any>()

    fun registerService(key: String, instance: Any) {
        if (!objMap.containsKey(key)) {
            objMap[key] = instance
        }
    }

    fun getService(key: String): Any? {
        return objMap[key]
    }

    fun reMove(key: String) {
        objMap.remove(key)
    }
}
